CONTRIBUTING.md

```diff
@@ -157,3 +158,32 @@ This will go through the entire source tree and update all of the references in
 - External dependencies. Right now there are only version files for client and data (eng\versioning\version_\[client|data\].txt) which only encompass the built binaries for their respective tracks. External dependencies for both client and data are next on the list which should allow modification of the parent/pom.xml to remove the list of version properties and dependency management sections which brings things one step closer to not having to publish the parent pom.
 - Management plane. Management is in the process of being moved to service pipeline builds. The versioning work needs to wait until that work is finished.
 - Service pipeline changes. The service pipelines currently have to build not only the libraries that are part of that pipeline but also the Azure SDK libraries that are dependencies. Once we GA and can start targeting the released version of those packages and pulling them from Maven instead of building them. An good example of this would be in sdk/appconfiguration/pom.service.xml where to build azure-data-appconfiguration we end up building azure-core, azure-core-test and azure-core-http-netty along with azure-data-appconfiguration instead of just building azure-data-appconfiguration.
+
```

Comment:
@JonathanGiles the changes in this file are for your comments in the original PR. The small novella below was added in response to your ask about describing what the process looks like.